src/mbgl/renderer/renderer_impl.cpp

```diff
@@ -140,20 +136,20 @@ void Renderer::Impl::render(const UpdateParameters& updateParameters) {
 
     // Remove removed images from sprite atlas.
     for (const auto& entry : imageDiff.removed) {
-        imageManager->removeImage(entry.first);
+        updateParameters.imageManager.removeImage(entry.first);
```

Comment:
What is weird for me about this is the ownership of this object is not clear. `imageManager` now lives on the same thread as the `Map` but is being accessed from the `Renderer` thread.

Reply:
(unlike the `FileSource` that is thread-safe)